### ISSUE TITLE
Mapper vs. DAO approach - Issue for Aspect Oriented Programming (AOP)

### ISSUE THREAD

**Issue by GoogleCodeExporter** [mybatis/old-google-code-issues#489]
```
What version of the MyBatis are you using? 3.0.6

Please describe the problem.  Unit tests are best!

I had implemented some SQL inserts using DAO and DAO.MyBatis implementation.  I 
had also implemented AOP to automatically
set updatedBy and updatedDate on all my model objects.  All of this worked fine 
including the AOP piece. I.e. AOP fired onPreInsert when doing the insert with 
a DAO/DAO mybatis implementation.

Then I ran into a situation which really made me rethink the DAO approach and 
instead use Mappers .. MapperFactoryBean .. Proxy etc.  I implemented this and 
ran into an issue with AOP .. it appears when using MapperFactoryBean .. that 
the MyBatis/Spring or perhaps just MyBatis .. not sure which ... isn't picking 
up on the insert .. so that when my @Before .. based on @Pointcut runs in 
synchronization with my MapperBean its not firing my @Before .. onPreInsert() 
.. method.  So my updatedBy and updatedDate objects aren't getting set.

In other words with the MapperFactorBean approach vs.SqlSessionDaoSupport the 
insert event isnt being picked up by AOP and my Pointcut isn't firing.

All I did in my AOP class was change the @Pointcut routine to look at 
com.fa.mapper.* instead of com.fa.dao.*.  This means that somehow the Proxy 
insert (or other CRUD) activity of the MapperFactoryBean isnt being detected by 
AOP.

Any and all thoughts would be appreciated.
What is the expected output? What do you see instead?


```

Original issue reported on code.google.com by `gpe...@gmail.com` on 6 Jan 2012 at 2:01
